infrastructure/general/setup-secure-wss.md

```diff
@@ -0,0 +1,89 @@
+
+Use the following commmand to generate a self-signed certificate using OpenSSL:
+
+--8<-- 'code/infrastructure/general/install-openssl.md'
+
+## Install a Proxy Server
+
+There are a lot of different implementations of a WebSocket proxy; some of the more widely used are [nginx](https://www.nginx.com/){target=_blank} and [apache2](https://httpd.apache.org/){target=_blank}, both of which are commonly used web server implementations. See the following section for configuration examples for both implementations:
+
+### Use NGINX
+
+1. Install the `nginx` web server: 
+    ```bash
+    apt install nginx
+    ```
+
+2. In an SSL-enabled virtual host add:
+    --8<-- 'code/infrastructure/general/nginx-config.md'
+
+3. Optionally, you can introduce some form of rate limiting:
+    --8<-- 'code/infrastructure/general/nginx-rate-limit.md'
+
+### Use Apache2
+
+Apache2 can run in various modes, including `prefork`, `worker`, and `event`. In this example, the [`event`](https://httpd.apache.org/docs/2.4/mod/event.html){target=_blank} mode is recommended for handling higher traffic loads, as it is optimized for performance in such environments. However, depending on the specific requirements of your setup, other modes like `prefork` or `worker` may also be appropriate.
+
+1. Install the `apache2` web server:
+    --8<-- 'code/infrastructure/general/install-apache2.md'
+
+2. The [`mod_proxy_wstunnel`](https://httpd.apache.org/docs/2.4/mod/mod_proxy_wstunnel.html){target=_blank} provides support for the tunneling of WebSocket connections to a backend WebSocket server. The connection is automatically upgraded to a WebSocket connection. In an SSL-enabled `virtualhost` add:
+    --8<-- 'code/infrastructure/general/apache2-config.md'
+
+    !!!warning "Older versions of `mod_proxy_wstunnel` don't upgrade the connection automatically and will need the following config added:"
+        ```apacheconf
+        RewriteEngine on
+        RewriteCond %{HTTP:Upgrade} websocket [NC]
+        RewriteRule /(.*) ws://localhost:9944/$1 [P,L]
+        RewriteRule /(.*) http://localhost:9944/$1 [P,L]
+        ```
+
+3. Optionally some form of rate limiting can be introduced:
+
+    ```bash
+    apt install libapache2-mod-qos
+    a2enmod qos
+    ```
+
+    And edit `/etc/apache2/mods-available/qos.conf`
+
+    ```conf
+    # allows max 50 connections from a single ip address:
+    QS_SrvMaxConnPerIP                                 50
+    ```
+
+## Connect to the Node
+
+1. Open [Polkadot.js Apps interface](https://polkadot.js.org/apps){target=_blank} and click the logo in the top left to switch the node
+2. Activate the **Development** toggle and input your node's address - either the domain or the IP address. Remember to prefix with `wss://` and if you're using the 443 port, append `:443` as follows:
+
+    ```bash
+    wss://example.com:443
+    ```
+
+![A sync-in-progress chain connected to Polkadot-JS UI](/images/infrastructure/general/maintain-wss.webp)
```

Comment:
```suggestion
![A sync-in-progress chain connected to Polkadot.js UI](/images/infrastructure/general/maintain-wss.webp)
```

Reply:
This image should be under `/images/infrastructure/general/setup-secure-wss/secure-wss-1.webp` and renamed to be `file-name-1.webp` - sometimes I shorten the file name if it's long